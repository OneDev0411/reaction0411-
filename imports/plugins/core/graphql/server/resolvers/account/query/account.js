import { transformIdFromBase64 } from "@reactioncommerce/reaction-graphql-utils";
import { userAccountQuery } from "/imports/accounts/server/methods/userAccountQuery";

export default function account(_, { id }) {
  // search for user from the Accounts collection via provided Account ID

/**
 * @name account
 * @method
 * @summary query the Accounts collection and return user account data
 * @param {Object} _ - an object containing the result returned from the resolver
 * @param {Object} args - an object containing Meteor context
 * @param {String} args.id - id of user to query
 * @param {Object} context - an object containing the per-request state
 * @return {Object} user account object
 */
export default function account(_, { id }, context) {
  // Trasform ID from base64
  // Returns an object. Use `.id` to get ID
  const idFromBase64 = transformIdFromBase64(id);

  // Pass Id into userAccountQuery function
  const userAccount = userAccountQuery(idFromBase64.id);

  // Return result of userAccountQuery()
  return userAccount;
}
