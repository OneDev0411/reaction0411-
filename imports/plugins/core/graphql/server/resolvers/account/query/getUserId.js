import { Meteor } from "meteor/meteor";

export default function getUserId(_, __, context) {
  const userId = context && context.user && context.user._id;
  if (!userId) throw new Meteor.Error("access-denied");
  return userId;
}
