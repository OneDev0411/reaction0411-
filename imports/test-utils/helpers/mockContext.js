const mockContext = {
  accountId: "FAKE_ACCOUNT_ID",
  appEvents: {
    emit() {},
    on() {}
  },
  collections: {},
  getAbsoluteUrl: jest.fn().mockName("getAbsoluteUrl").mockImplementation((path) => {
    const adjustedPath = path[0] === "/" ? path : `/${path}`;
    return `https://app.mock${adjustedPath}`;
  }),
  getFunctionsOfType: jest.fn().mockName("getFunctionsOfType").mockReturnValue([]),
  mutations: {},
  queries: {},
  userHasPermission: jest.fn().mockName("userHasPermission"),
  userId: "FAKE_USER_ID"
};

/**
 * @summary Returns a mock collection instance with the given name
 * @param {String} collectionName The collection name
 * @returns {Object} Mock collection instance
 */
export function mockCollection(collectionName) {
  return {
    insert() {
      throw new Error("insert mongo method is deprecated, use insertOne or insertMany");
    },
    remove() {
      throw new Error("remove mongo method is deprecated, use deleteOne or deleteMany");
    },
    update() {
      throw new Error("update mongo method is deprecated, use updateOne or updateMany");
    },
    bulkWrite: jest.fn().mockName(`${collectionName}.bulkWrite`).mockReturnValue(Promise.resolve({
      nMatched: 2,
      nModified: 2,
      result: {
        writeErrors: []
      }
    })),
    deleteOne: jest.fn().mockName(`${collectionName}.deleteOne`).mockReturnValue(Promise.resolve({
      deletedCount: 1
    })),
    deleteMany: jest.fn().mockName(`${collectionName}.deleteMany`),
    find: jest
      .fn()
      .mockName(`${collectionName}.find`)
      .mockReturnThis(),
    findOne: jest.fn().mockName(`${collectionName}.findOne`),
    findOneAndDelete: jest.fn().mockName(`${collectionName}.findOneAndDelete`),
    findOneAndUpdate: jest.fn().mockName(`${collectionName}.findOneAndUpdate`),
    fetch: jest.fn().mockName(`${collectionName}.fetch`),
    insertOne: jest.fn().mockName(`${collectionName}.insertOne`),
    insertMany: jest.fn().mockName(`${collectionName}.insertMany`),
    toArray: jest.fn().mockName(`${collectionName}.toArray`),
    updateOne: jest.fn().mockName(`${collectionName}.updateOne`).mockReturnValue(Promise.resolve({
      matchedCount: 1,
      modifiedCount: 1
    })),
    updateMany: jest.fn().mockName(`${collectionName}.updateMany`)
  };
}

[
  "Accounts",
  "Assets",
  "Cart",
  "Catalog",
  "Emails",
  "ExampleIOUPaymentRefunds",
  "Groups",
  "MediaRecords",
  "NavigationItems",
  "NavigationTrees",
  "Notifications",
  "Orders",
  "Packages",
  "Products",
  "Revisions",
  "roles",
  "SellerShops",
  "Shipping",
  "Shops",
  "SimpleInventory",
  "Tags",
  "Templates",
  "Themes",
  "users"
].forEach((collectionName) => {
  mockContext.collections[collectionName] = mockCollection(collectionName);
});

mockContext.collections.Media = {
  find: jest.fn().mockName("Media.find"),
  findLocal: jest.fn().mockName("Media.findLocal"),
  findOne: jest.fn().mockName("Media.findOne"),
  findOneLocal: jest.fn().mockName("Media.findOneLocal"),
  insert: jest.fn().mockName("Media.insert"),
  update: jest.fn().mockName("Media.update"),
  remove: jest.fn().mockName("Media.remove")
};

export default mockContext;
