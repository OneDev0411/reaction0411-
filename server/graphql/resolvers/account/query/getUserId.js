export default function getUserId(_, __, context) {
  const userId = context && context.user && context.user._id;
  if (!userId) throw new Error("access-denied");
  return userId;
}
