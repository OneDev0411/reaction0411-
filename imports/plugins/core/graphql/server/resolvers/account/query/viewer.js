import account from "./account";

export default function viewer(_, __, context) {
  const userId = (context && context.user && context.user._id) || {};
  return account(_, { _id: userId }, context)
}
