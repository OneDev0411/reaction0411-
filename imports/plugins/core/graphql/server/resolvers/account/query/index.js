import account from "./account";
import administrators from "./administrators";
import group from "./group";
import groups from "./groups";
import viewer from "./viewer";

export default {
  account,
  administrators,
  group,
  groups,
  viewer
};
