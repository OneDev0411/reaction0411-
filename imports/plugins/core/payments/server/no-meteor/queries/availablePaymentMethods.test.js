import Factory from "/imports/test-utils/helpers/factory";
import mockContext from "/imports/test-utils/helpers/mockContext";
import query from "./availablePaymentMethods";

jest.mock("/imports/plugins/core/core/server/no-meteor/pluginRegistration", () => ({
  paymentMethods: {
    mockPaymentMethod: {
      name: "mock",
      displayName: "Mock!"
    }
  }
}));

const fakeShop = Factory.Shop.makeOne();
const mockShopById = jest.fn().mockName("shopById");

beforeAll(() => {
  mockContext.queries = {
    shopById: mockShopById
  };
});

beforeEach(() => {
  jest.resetAllMocks();
  mockShopById.mockClear();
});

test("throws if shop not found", async () => {
  mockContext.userHasPermission.mockReturnValueOnce(true);
  mockShopById.mockReturnValueOnce();

  await expect(query(mockContext, "nonexistent-shop-id")).rejects.toThrowErrorMatchingSnapshot();
  expect(mockShopById).toHaveBeenCalledWith(mockContext, "nonexistent-shop-id");
});

test("returns empty array when shop has no valid payment methods", async () => {
  mockShopById.mockReturnValueOnce(fakeShop);
  mockContext.userHasPermission.mockReturnValueOnce(true);

  const result = await query(mockContext, mockContext.shopId);
  expect(mockShopById).toHaveBeenCalledWith(mockContext, mockContext.shopId);
  expect(result).toEqual([]);
});

test("returns available payment methods for a shop", async () => {
  mockShopById.mockReturnValueOnce(fakeShop);
  mockContext.userHasPermission.mockReturnValueOnce(true);
  fakeShop.availablePaymentMethods.push("mockPaymentMethod");

  const result = await query(mockContext, mockContext.shopId);
  expect(mockShopById).toHaveBeenCalledWith(mockContext, mockContext.shopId);
  expect(result).toEqual([{
    name: "mock",
    displayName: "Mock!",
    isEnabled: true
  }]);
});
