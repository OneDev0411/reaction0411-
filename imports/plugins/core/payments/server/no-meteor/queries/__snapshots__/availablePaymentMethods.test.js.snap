// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws if shop not found 1`] = `"ReactionError is not defined"`;
